MetodeDeDezvoltareSoftware/MDS.md

```diff
@@ -171,103 +169,25 @@ cerințe in schimbare | cerințe relativ fixe | cerințe limitate
 echipe mici | echipe mari | echipe mici
 cultură orientată spre schimbare | cultură orientată spre ordine | cultură orientată spre calitate și precizie
 
-#Z
-
-> **AICI NU SE AFLA DECAT CHESTIILE CE ERAU MARCATE CU "DE RETINUT" IN CURS.**
-
-Este bazat pe:
-- logica de ordin 1, cu predicate
-- teoria multimiilor
-- notatii auxiliare
-
-##Specificatii formale
-O specificație formală:
--folosește notații matematice pentru a descrie într-un model precis ce proprietăți trebuie să aibă un sistem
--descrie ce trebuie să facă sistemul și nu cum
--independent de cod
--poate fi folosită pentru înțelegerea cerințelor și analiza lor (uneori se poate si genera cod dintr-o specificație suficient de precisă)
-
-În Z descompunerea unei specificații se face în mai multe piese numite *scheme*.
-
-##Operatori logici
-- $\neg$ - negatie
-- $\wedge$- conjunctie
-- $\vee$- disjunctie
-- $\implies$ - implicatie
-- $\iff$ - echivalenta
-
-##Egalitate
-- $=$ - egalitate
-
-##Cuantificare
-$Q x_1: S_1 ; \dots x_n : S_n | p \bullet q$ unde $Q$ este $\forall, \exists, \exists_1$
-Sensul este $\forall x_1: S_1 ; \dots x_n : S_n ( p \implies q)$
-
-##Multimi
-
-###Multimi prin enumerare
-$\{ e_1, \dotsc, e_n\}$ unde elementele au tipuri compatibile
-
-###Multimi definite prin proprietati
-$\{ x: T | pred(x) \bullet expr(x) \}$ reprezinta toate elementele care rezulta evaluand $expr(x)$ pentru toti $x$ de tip $T$ care starisfac $pred(x)$.
-
-###Operatii pe multimi
-- apartenenta
-- submultime
-- Operatorul de generare a submultimilor
-- Produs cartezian
-- reuniune
-- Intersectie
-- Diferenta multimilor
-
-##Tipuri
-Definitii de noi tipuri: `weekDay ::= mon | tue | wed | thu | fri | sat | sun`
-
-##Variabile
-Se decalara cu $x:\mathbb{Z}$.
-
-###Relatii
-####Relatii pe tipuri de multimi
-$S \leftrightarrow T$ este multimea relatiilor intre multimiile $S$ si $T$
-
-$a \rightarrowtail b$ denota perechea $(a, b)$, daca $(a, b) \in S \leftrightarrow T $
-
-####Operatii pe relatii
-- Domeniu $\text{domR} = \{ a : S; b : T | a \rightarrowtail b \in R \bullet a \}$
-- Codomeniu $\text{ranR} = \{ a : S; b : T | a \rightarrowtail b \in R \bullet b \}$
-- Relatia inversa
-- Compunerea
-- Inchideri
-
-##Functii
-
-###Notatii
-In loc de $\leftrightarrow$ folosim notatiile:
-- $\rightarrow$ pentru functii totale
-- $\nrightarrow$ pentru functii partiale
 
 #JML
 
-E un limbaj de specificație formală pentru Java. Comentarii in codul sursa descriu formal cum trebuie sa se comporte un modul, prevenind ambiguitatea. Se foloseste de e invarianți, pre- și postcondiții.
+E un limbaj de specificație formală pentru Java. Comentarii in codul sursa descriu formal cum trebuie sa se comporte un modul, prevenind ambiguitatea. Se foloseste de invarianți, pre- și postcondiții.
 
-```
-public class ATM
-{
+```java
```

Comment:
Keep the `java` to mark the language. This is useful for syntax highlighting when exporting.